public/plugin.tsx

```diff
@@ -387,38 +394,58 @@ export class ObservabilityPlugin
     coreRefs.overlays = core.overlays;
 
     const { dataSourceService, dataSourceFactory } = startDeps.data.dataSources;
+    dataSourceFactory.registerDataSourceType(S3_DATA_SOURCE_TYPE, S3DataSource);
 
     // register all s3 datasources
-    const registerS3Datasource = () => {
-      dataSourceFactory.registerDataSourceType(S3_DATASOURCE_TYPE, S3DataSource);
+    const registerDataSources = () => {
       core.http.get(`${DATACONNECTIONS_BASE}`).then((s3DataSources) => {
         s3DataSources.map((s3ds) => {
           dataSourceService.registerDataSource(
-            dataSourceFactory.getDataSourceInstance(S3_DATASOURCE_TYPE, {
+            dataSourceFactory.getDataSourceInstance(S3_DATA_SOURCE_TYPE, {
+              id: htmlIdGenerator(OBS_S3_DATA_SOURCE)(DATA_SOURCE_TYPES.S3Glue),
               name: s3ds.name,
               type: s3ds.connector.toLowerCase(),
-              metadata: s3ds,
+              metadata: {
+                ...s3ds.properties,
+                ...{
```

Comment:
why is it spreading an object?

Reply:
include any other metadata in s3 properties

Reply:
comment is for line 410, not 409